src/main/scala/zio/prelude/Associative.scala

```diff
@@ -76,102 +158,228 @@ object Associative extends Lawful[AssociativeEqual] {
   ): Associative[F[A]] =
     derive.derive(associative)
 
-  implicit val DoubleProdAssociative: Associative[Prod[Double]] =
-    make[Prod[Double]]((l, r) => Prod(l * r))
+  /**
+   * The `Commutative` and `Identity` instance for the product of `Double`
+   * values.
+   */
+  implicit val DoubleProdCommutativeIdentity: Commutative[Prod[Double]] with Identity[Prod[Double]] =
```

Comment:
will we be as bold as to declare `Inverse[Prod[Double]]` (and for other `Prod`s for numbers)? :laughing: :pirate_flag:

Reply:
I think so. We have already said we are going to provide instances for floating point values despite potential rounding errors so I don't see why this would be any different.

Reply:
Just for the record, what we did here: the inverse for multiplication is division and that can mean _division by zero_.
the `Inverse[Prod[_]]` instances for `Float` and `Double` are fine, because division by both `+0.0` and `-0.0` are defines (it's one of the `NaN`s, but that is still valid _IEEE 754_ `Float`/`Double`).
Instances for `Byte`, `Short`, `Int` and `Long`, maybe not, there can happen division by `0` and that will throw. But maybe that's fine and pragmatic solution. Or maybe not and we're being to wild :pirate_flag: :smile_cat: 

I just want to make sure we're both on the same page :wink:

Reply:
That is a good point. I think floating point errors are okay but division by zero is not. I will revert these.